cloudbuild.yaml

```diff
@@ -0,0 +1,31 @@
+# Copyright 2021 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+steps:
+  - name: "docker"
+    args:
+      - build
+      - --tag=gcr.io/$PROJECT_ID/opentelemetry-operations-python-e2e-test-server
+      - --file=e2e-test-server/Dockerfile
+      - .
+
+  - name: gcr.io/opentelemetry-ops-e2e/opentelemetry-operations-e2e-testing:0.5.5
```

Comment:
Nit: Do we want the name that shows up to be this, or just "Push image to GCR"

Reply:
So `name` is (confusingly) the container image to run: https://cloud.google.com/build/docs/build-config#name. This step runs the test server, I'm not pushing the built image to GCR at all actually.

I think [`id`](https://cloud.google.com/build/docs/build-config#id) changes what shows in the UI, but it also used to configure dependencies. Lets me check

Reply:
Updated. I used kebab strings for ID so its clear they are not primarily human readable. Now the build UI looks like this

<img width="515" alt="Screen Shot 2021-05-03 at 2 11 22 PM" src="https://user-images.githubusercontent.com/1510004/116915199-b96cb080-ac19-11eb-8a59-e51a89e41136.png">

Reply:
Ah, that looks good.